postgresql-16-upgrade/scripts/restore.sh

```diff
@@ -0,0 +1,292 @@
+  exit 1
+fi
+
+# --- Main
+trap _exit EXIT
+trap 'exit 1' ERR
+
+if [[ "$FORCE" == "false" ]]; then
+  echo_step "Check backup status"
+  status=$(kubectl $KUBECTL_ARGS get configmap postgresql-backup --output jsonpath="{.data.status}")
+  if [[ "$status" == "" ]]; then
+    echo_error "Failed to retrieve the backup status from 'postgresql-backup"
+    exit 1
+  fi
+  echo_ok
+fi
+
+if [[ "$FORCE" == "true" ]] || [[ "$status" == "completed" ]]; then
+  echo_step "Check backup file"
+  secret_hash=$(kubectl $KUBECTL_ARGS get configmap postgresql-backup --output jsonpath="{.data.hash}")
+  if [[ -z "$secret_hash" ]]; then
+    echo "Unable to retrieve the backup file hash from postgresql-backup configMap"
+    exit 1
+  fi
+  if [[ "$OS" == "darwin" ]]; then
+    hash=$(md5 -q $INPUT_FILE)
+  else
+    hash=$(md5sum $INPUT_FILE | awk '{print $1}')
+  fi
+
+  if [[ "$secret_hash" != "$hash" ]]; then
+    echo "Checksum mismatch error"
+    echo "expected file hash : ${secret_hash}"
+    echo "provided file hash : ${hash}"
+    exit 1
+  fi
+  echo_ok
+
+  echo_step "Scale down GitGuardian deployments"
+  deployments=()
+  readarray -t deployments < <(kubectl $KUBECTL_ARGS get deployment \
+      --selector="kots.io/app-slug=${REPLICATED_APP}, app.kubernetes.io/name=${REPLICATED_APP}" \
+      --output=name 2>$ERROR_LOG_FILE)
+  if [[ "${#deployments[@]}" -gt 0 ]]; then
+    for deployment in "${deployments[@]}"; do
+      # Get the current replicas
+      replicas=$(kubectl $KUBECTL_ARGS \
+        get $deployment \
+        --output jsonpath="{.status.replicas}")
+      # Scale down
+      kubectl $KUBECTL_ARGS \
+        scale $deployment \
+        --timeout=3m \
+        --replicas=0 2>$ERROR_LOG_FILE
+      # Add replicas annotation
+      kubectl $KUBECTL_ARGS \
+        annotate --overwrite $deployment \
+        replicas="${replicas}" >/dev/null 2>$ERROR_LOG_FILE
+    done
+  fi
+  echo_ok
+
+  echo_step "Retrieve PostgreSQL pod"
+  postgres_pod_selector="kots.io/app-slug=${REPLICATED_APP}, app.kubernetes.io/name=postgresql, app.kubernetes.io/component=primary"
+  pod=$(kubectl $KUBECTL_ARGS get pod \
+    --field-selector="status.phase=Running" \
+    --selector="$postgres_pod_selector" \
+    --output=name 2>$ERROR_LOG_FILE | head -1)
+  if [[ -z "$pod" ]]; then
+    echo "No running PostgreSQL pod found" >$ERROR_LOG_FILE
+    exit 1
+  fi
+  echo_ok
+
+  echo_step "Restore the GitGuardian DB"
+  postgres_restore_cmd="PGPASSWORD=\$POSTGRES_POSTGRES_PASSWORD psql -U postgres"
+  gunzip -c ${INPUT_FILE} | \
+    kubectl $KUBECTL_ARGS \
+      exec -i $pod -- \
+      bash -c "$postgres_restore_cmd" >/dev/null 2>$ERROR_LOG_FILE
+  if [[ -s "$ERROR_LOG_FILE" ]]; then
+    exit 1
+  fi
+  echo_ok
+
+  echo_step "Update backup status"
+  kubectl $KUBECTL_ARGS \
+    patch configmap postgresql-backup \
+    --type=json \
+    --patch='[{ "op": "replace", "path": "/data/status", "value": "restored" }]' >/dev/null 2>$ERROR_LOG_FILE
```

Comment:
the CM contains a date field, it is not clear to user it matches to:
- backup (this is the case)
- restore

Should we rename that field ? Does having 2 fields (backup_date, restore_date) make sense?

Reply:
Actually, this field could be removed as It is not read by any scripts, It is purely informational, so I suggest removing it.